### ISSUE TITLE
Name and format of modlist and modliststring

### ISSUE THREAD

**Issue by maelle** [cvitolo/fuse#5]
This issue is part of [this JOSS review](https://github.com/openjournals/joss-reviews/issues/52)

These are only suggestions and questions:

* I'd be in favour of naming `modlist` more explicitely, e.g. `fuse_models`, and maybe even without "list" in the name since it's a `data.frame` (I understand the reason for calling it list, but I guess that because list is an object class in R it might be confusing?).

* I do not understand the use of `modliststring`. When looking for its occurrences in the repo I could only find the documentation. If the only role of `modliststring` is to explain the codes, why are abbreviations used? Maybe it'd make sense to no longer have `modliststring` and instead have `modlist` and a data.frame per type of code with a column "code" and a column "meaning", or to have only `modlist` and its documentation. Or to have only `modliststring` and the documentation, i.e. use the abbreviations instead of the codes, and explain the abbrevations in the documentation?

* The names of the variables themselves could be a bit more explicit, e.g. `esoil` could be `soil_evaporation`. Unless those are all abbreviations commonly used in hydrology?

**Comment by maelle** [comment]
"On Clark's model list, the parent models (Topmodel, VIC, PRMS and Sacramento) have id = 60, 230, 342 and 426 respectively." -> why not add a variable `clark_name`?

**Comment by maelle** [comment]
I see you use the codes (and not the abbreviations) internally, e.g. in 

```r
  model <- data.frame(matrix(readmd2var(mid,number=TRUE),ncol=8,nrow=1))
names(model) <- names(readmd2var(mid,number=TRUE))
```

However for the user codes are not really useful, are they?

I think it's great to have a function for showing values for a single model ID only, but why have two of them, i.e. `readmd2var` and `readmd`? You could export only one with focus on "pretty" (after all otherwise the user can filter from `modlist` or `modliststring`) and not export the other one which you use internally?

**Comment by maelle** [comment]
Oh I see there's `fuseInfo` too. It doesn't seem easy to know which one to use, the three `fuseInfo`, `readmd` and `readmd2var` do something really similar, right?

**Comment by cvitolo** [comment]
The names I used in the package reflect the original names in the fortran code. Keeping consistent names for the various objects allow me to track the changes between the fortran and R code.
For this reason I cannot change the names of `modlist` or variables (e.g. esoil).

I need to keep modliststring because this is the list used in the fortran code. I had to convert the strings into numbers (modlist) because R handles strings very slowly. Keeping both modlist and modliststring allows me to look-up what string corresponds to what number. Hope this makes sense.

I cannot create a `clark_name` because there are names only for 4 models (which are considered parent models).

readmd2var, readmd and fuseInfo have different goals:
- readmd2var is used by the model itself to generate the model structure from the id number (mid)
- readmd is used for reporting, to generate tables (usually via xtable) to list model structures and describe each component.
- fuseInfo is used to generate tables to summarise which model component and parameters are used given the model id. This is important for calculating the posterior distribution of parameter values and model structures.
I have added these comment to the related documentation.
